Formula/g/gnmic.rb

```diff
@@ -0,0 +1,28 @@
+class Gnmic < Formula
+  desc "GNMI CLI client and collector"
+  homepage "https://gnmic.openconfig.net"
+  url "https://github.com/openconfig/gnmic.git",
+      tag:      "v0.36.2",
+      revision: "a7844a6d7d47c302df1fabb845dd22e4e4b70e22"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CGO_ENABLED"] = "0"
+    ldflags = %W[
+      -s -w
+      -X github.com/openconfig/gnmic/pkg/app.version=#{version}
+      -X github.com/openconfig/gnmic/pkg/app.commit=#{Utils.git_head}
+      -X github.com/openconfig/gnmic/pkg/app.date=#{time.iso8601}
+      -X github.com/openconfig/gnmic/pkg/app.gitURL=https://github.com/openconfig/gnmic
+    ]
+    system "go", "build", *std_go_args(ldflags:)
+  end
+
+  test do
+    version_output = shell_output(bin/"gnmic version 2>&1")
+    assert_match "version : #{version}", version_output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I've updated the test to attempt a gNMI connection to localhost which will fail predictably. There is no real way to test as the program would need a running gNMI application somehwere.